packages/catalog-process/src/services/catalogService.ts

```diff
@@ -488,7 +488,9 @@ export function catalogServiceBuilder(
       const eservice = await retrieveEService(eserviceId, readModelService);
       assertRequesterAllowed(eservice.data.producerId, authData.organizationId);
 
-      if (eservice.data.descriptors.length > 0) {
+      if (
+        eservice.data.descriptors.some((d) => d.state !== descriptorState.draft)
+      ) {
         throw eServiceCannotBeDeleted(eserviceId);
       }
```

Comment:
Possible refactor: we could just use `assertIsDraftEservice` which contains both the check and the error throwing. The only difference would be which error is thrown. Indeed, `assertIsDraftEservice` throws an `eserviceNotInDraftState` error.

Reply:
IMO it makes a lot of sense

Reply:
Updated, both here and in `updateEService`, because it's an identical situation.